### PR TITLE
ENH: initial vectors in sparse.linalg.minres

### PR DESCRIPTION
**Pull request by ashwinpathak20**
In reference with the issue #6843, to intend to use the input x and not deprecate x0, in line 94,
r1 = b - Ax seems the correct solution. Please have a look and suggest me appropriate changes if any.

### CODE REVIEW

File: scipy/sparse/linalg/isolve/minres.py [hunk 1]
```diff
@@ -85,14 +85,13 @@ def minres(A, b, x0=None, shift=0.0, tol=1e-5, maxiter=None, xtype=None,
 
     eps = finfo(xtype).eps
 
-    x = zeros(n, dtype=xtype)
```
Comment (perimosocordiae) [inline scipy/sparse/linalg/isolve/minres.py]: The style checker is complaining that there are too many blank lines here. Could you delete line 87 as well, please?

Comment (ashwinpathak20) [inline scipy/sparse/linalg/isolve/minres.py]: @perimosocordiae I removed the line still the CI failed. Can you tell me what is the issue?

Comment (perimosocordiae) [inline scipy/sparse/linalg/isolve/minres.py]: The linting error is fixed, but one of the test runs is segfaulting. I'm not sure if this is related to your changes or not.

Comment (pv) [inline scipy/sparse/linalg/isolve/minres.py]: The segfault is unrelated, probably some bug in Numpy master.

**Comment by pv** [comment]
needs tests

**Comment by ashwinpathak20** [comment]
@pv What kinds of tests should I include in this one? and where?

**Comment by MichaelMarien** [comment]
Seems like this issue is blocked due to tests only? I saw from pv's comment in https://github.com/scipy/scipy/pull/6887 that it's ok to continue work on dormant issues started by others. I'll add some tests to get this fixed asap.

**Comment by rgommers** [comment]
@MichaelMarien that would be great, thanks. I'd suggest to take over these comments, rebase on master, and your changes on top and open a new PR.

**Comment by rgommers** [comment]
continued in gh-10719, closing this PR. thanks @ashwinpathak20

**Pull request closed without merging**